sched/semaphore/sem_post.c

```diff
@@ -85,19 +87,54 @@ int nxsem_post_slow(FAR sem_t *sem)
 
   flags = enter_critical_section();
 
-  /* Check the maximum allowable value */
+  if (mutex)
+    {
+      /* Mutex post from interrupt context is not allowed */
+
+      DEBUGASSERT(!up_interrupt_context());
+
+      /* Lock the mutex for us by setting the blocking bit */
+
+      mholder = atomic_fetch_or(NXSEM_MHOLDER(sem), NXSEM_MBLOCKS_BIT);
+
+      /* Mutex post from another thread is not allowed, unless
+       * called from nxsem_reset
```

Comment:
You could just spoof the ownership directly from nxmutex_reset, making most of these sanity checks redundant.